LanguageFeatures/Extension-types/static_analysis_extension_types_A10_t14.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A compile-time error occurs if an extension type has a
+/// non-extension superinterface whose transitive alias expansion is a type
+/// variable, a deferred type, any top type (including dynamic and void), the
+/// type Null, any function type, the type Function, any record type, the type
+/// Record, or any type of the form T? or FutureOr<T> for any type T.
+///
+/// @description Checks that it is not an error if an representation type of an
+/// extension type is a  top type, the type `Null`, a function type, the type
+/// `Function`, a record type, the type `Record`, a type of the form `T?` or
+/// `FutureOr<T>`
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "dart:async" show FutureOr;
+import "../../Utils/expect.dart";
+import "../../Utils/static_type_helper.dart";
+
+typedef void Foo();
+
+extension type ET1(dynamic id) {}
+
+extension type ET2(Object id) {}
+
+extension type ET3(Object? id) {}
+
+extension type ET4(Null id) {}
+
+extension type ET5(Never id) {}
```

Comment:
This one should be a compile-time error, based on https://github.com/dart-lang/language/pull/3393. I guess I forgot to mention that PR, sorry about that.

Reply:
We should probably have a separate test where it's confirmed that we get an error for a representation type which is `Never`, `X extends Never`, etc., as described in https://github.com/dart-lang/language/blob/main/accepted/2.12/nnbd/feature-specification.md#helper-predicates about `BOTTOM(_)`.

Reply:
Thanks!